MaxText/inference_microbenchmark_sweep.py

```diff
@@ -0,0 +1,156 @@
+    - ar_key_axis_order_list: comma delimited string of ar_key_axis_order
+    - ar_value_axis_order_list: comma delimited string of ar_value_axis_order
+    - accelerator: name of the accelerator
+    - flatten_microbenchmark_results: Whether or not to flatten results. Should
+      be true
+  """
+  pyconfig.initialize(sys.argv)
+  config = pyconfig.config
+  base_run_name = config.run_name
+
+  with open(config.inference_metadata_file, encoding='utf-8') as json_file:
+    inference_metadata = json.load(json_file)
+    print(f"inference_metadata: {inference_metadata}")
+
+  key_value_axis_order_product_id_list = inference_metadata['key_value_axis_order_product_id_list'].split(':')
+  prefill_key_axis_order_list = inference_metadata['prefill_key_axis_order_list'].split(':')
+  prefill_value_axis_order_list = inference_metadata['prefill_value_axis_order_list'].split(':')
+  ar_key_axis_order_list = inference_metadata['ar_key_axis_order_list'].split(':')
+  ar_value_axis_order_list = inference_metadata['ar_value_axis_order_list'].split(':')
+
+  start_key_value_axis_order_product_id = key_value_axis_order_product_id_list[0]
+  end_key_value_axis_order_product_id = key_value_axis_order_product_id_list[-1]
+
+  results = []
+  for (
+    key_value_axis_order_product_id,
+    prefill_key_axis_order,
+    prefill_value_axis_order,
+    ar_key_axis_order,
+    ar_value_axis_order,
+  ) in zip(
+    key_value_axis_order_product_id_list,
+    prefill_key_axis_order_list,
+    prefill_value_axis_order_list,
+    ar_key_axis_order_list,
+    ar_value_axis_order_list,
+  ):
+    print(f"key_value_axis_order_product_id {key_value_axis_order_product_id}")
+    print(f"prefill_key_axis_order {prefill_key_axis_order}")
+    print(f"prefill_value_axis_order {prefill_value_axis_order}")
+    print(f"ar_key_axis_order {ar_key_axis_order}")
+    print(f"ar_value_axis_order {ar_value_axis_order}")
+
+    run_tag = (
+      f"{key_value_axis_order_product_id}-{prefill_key_axis_order.replace(',','')}-{ar_key_axis_order.replace(',','')}"
+    )
+    run_name = f"{base_run_name}/{run_tag}"
+
+    tensorboard_dir = os.path.join(config.base_output_directory, run_name, "tensorboard", "")
+    pyconfig._config.keys['prefill_key_axis_order'] = prefill_key_axis_order # pylint: disable=protected-access
+    pyconfig._config.keys['prefill_value_axis_order'] = prefill_value_axis_order # pylint: disable=protected-access
+    pyconfig._config.keys['ar_key_axis_order'] = ar_key_axis_order # pylint: disable=protected-access
+    pyconfig._config.keys['ar_value_axis_order'] = ar_value_axis_order # pylint: disable=protected-access
+    pyconfig._config.keys['tensorboard_dir'] = tensorboard_dir # pylint: disable=protected-access
+    pyconfig._config.keys['run_name'] = run_name # pylint: disable=protected-access
+    max_utils.write_config_raw_keys_for_gcs(pyconfig._config.keys) # pylint: disable=protected-access
```

Comment:
You can skip this since there is already
```
--save_config_to_gcs=True
```
in the [dags/inference/configs/maxtext_inference_microbenchmark_gce_config.py](https://github.com/GoogleCloudPlatform/ml-auto-solutions/pull/288/files#diff-0ca160e04db7871a01e5f1aafc9705f3f52efe9b1321b2901289554e806f55d9)

Reply:
My understanding is that `write_config_raw_keys_for_gcs` is called once during `initialize` (https://github.com/google/maxtext/blob/main/MaxText/pyconfig.py#L225), and writes the default values of prefill and ar axis orders to GCS. And so each time we loop for a different prefill/ar axis, we need to explicitly call `max_utils.write_config_raw_keys_for_gcs` again to make sure we right the updated values to GCS. 

Did you take this out in your code, and it worked for you?

Reply:
@morgandu should I do a test without `max_utils.write_config_raw_keys_for_gcs`?